x-pack/test/security_solution_endpoint_api_int/apis/resolver/entity_id.ts

```diff
@@ -0,0 +1,156 @@
+          index: genData.indices[0],
+          body: {
+            query: {
+              bool: {
+                filter: [
+                  {
+                    term: { 'process.parent.entity_id': origin.process.parent!.entity_id },
+                  },
+                ],
+              },
+            },
+          },
+        });
+        const { body }: { body: ResolverEntityIndex } = await supertest.get(
+          // using the same indices value here twice to force the query parameter to be an array
+          // for some reason using supertest's query() function doesn't construct a parsable array
+          `/api/endpoint/resolver/entity?_id=${res.body.hits.hits[0]._id}&indices=${eventsIndexPattern}&indices=${eventsIndexPattern}`
+        );
+        expect(body).to.be.empty();
+      });
+    });
+
+    describe('children', () => {
+      let origin: Event;
+      let childNoEntityID: Event;
+      let childWithEntityID: Event;
+      let events: Event[];
+      let genData: InsertedEvents;
+
+      before(async () => {
+        // construct a tree with an origin and two direct children. One child will not have an entity_id. That child
+        // should not be returned by the backend.
+        origin = generator.generateEvent({ entityID: 'a' });
+        childNoEntityID = generator.generateEvent({
+          parentEntityID: origin.process.entity_id,
+          ancestry: [origin.process.entity_id],
+        });
+        // force it to be empty
+        childNoEntityID.process.entity_id = '';
+
+        childWithEntityID = generator.generateEvent({
+          entityID: 'b',
+          parentEntityID: origin.process.entity_id,
+          ancestry: [origin.process.entity_id],
+        });
+        events = [origin, childNoEntityID, childWithEntityID];
+        genData = await resolver.insertEvents(events);
+      });
+
+      after(async () => {
+        await resolver.deleteData(genData);
+      });
+
+      it('does not find children without a process entity_id', async () => {
+        const { body }: { body: ResolverTree } = await supertest
+          .get(`/api/endpoint/resolver/${origin.process.entity_id}`)
+          .expect(200);
+        expect(body.children.childNodes.length).to.be(1);
+        expect(body.children.childNodes[0].entityID).to.be(childWithEntityID.process.entity_id);
+      });
+    });
+
+    describe('ancestors', () => {
+      let origin: Event;
+      let ancestor1: Event;
+      let ancestor2: Event;
+      let ancestorNoEntityID: Event;
+      let events: Event[];
+      let genData: InsertedEvents;
+
+      before(async () => {
+        // construct a tree with an origin that has two ancestors. The origin will have an empty string as one of the
+        // entity_ids in the ancestry array. This is to make sure that the backend will not query for that event.
+        ancestor2 = generator.generateEvent({
+          entityID: '2',
+        });
+        ancestor1 = generator.generateEvent({
```

Comment:
having trouble following what `ancestor1` is for. but i'm tired. maybe you can explain it to me next week :)

Reply:
Yeah it was more of a sanity check. The origin node in that test has its `process.parent.entity_id` set to ancestor1's `process.entity_id`, but the backend should be using the `ancestry` array to determine the ancestors if it exists. So since ancestor1 was not returned in the request we can be assured that the backend is properly using the ancestry array.